components/prism-fsharp.js

```diff
@@ -0,0 +1,29 @@
+Prism.languages.fsharp = Prism.languages.extend('clike', {
+	'comment': [
+		{
+			pattern: /(^|[^\\])\(\*[\w\W]*?\*\)/g,
+			lookbehind: true
+		},
+		{
+			pattern: /(^|[^\\:])\/\/.*/g,
+			lookbehind: true
+		}
+	],
+	'keyword': /\b(abstract|and|as|assert|base|begin|class|default|delegate|do|done|downcast|downto|elif|else|end|exception|extern|false|finally|for|fun|function|global|if|in|inherit|inline|interface|internal|lazy|let|let!|match|member|module|mutable|namespace|new|not|null|of|open|or|override|private|public|rec|return|return!|select|static|struct|then|to|true|try|type|upcast|use|use!|val|void|when|while|with|yield|yield!|asr|land|lor|lsl|lsr|lxor|mod|sig|atomic|break|checked|component|const|constraint|constructor|continue|eager|event|external|fixed|functor|include|method|mixin|object|parallel|process|protected|pure|sealed|tailcall|trait|virtual|volatile)\b/g,
+	'string': /@?("""|"|')((\\|\n)?.)*?\1(B)?/g,
+	'preprocessor': /^\s*#.*/gm,
+	'number': [
+		{
+			pattern: /\b-?(0x((\d*)|([a-fA-F]*))*(un|LF)?)\b/g
+		},
+		{
+			pattern: /\b-?(0b((0|1))*(y|uy)?)\b/g
+		},
+		{
+			pattern: /\b-?(\d*\.(f|\d)?\d*(f|F|m|M|((E|e)(\+|-)?\d*))?)\b/g
+		},
+		{
+			pattern: /\b-?(\d{1,}(y|uy|s|us|l|u|ul|L|UL|I)?)\b/g
+		}
+	]
```

Comment:
I think you should simplify the 4 patterns here:
1st: `((\d*)|([a-fA-F]*))*` => `[\da-fA-F]*`. And there seems to be a pair of parentheses that can be removed.
2nd: `((0|1))*` => `[01]*`. There is also a pair of parentheses that's not needed.
3rd: I guess the `f` in the middle is for handling the numers ending with a dot? What about... `2.F` or `2.m`? Are they valid numbers in F#? Furthermore, `(E|e)` should be simplified as `[Ee]` and `f|F|m|M` as `[fFmM]`. And still a pair of unneeded parentheses.
4th: `\d{1,}` => `\d+`. An the pair of unneeded parentheses can be removed.

Try to keep the patterns as short as possible. This will actually make them more readable. Counting the parentheses is probably the hardest part in understanding a regexp ^^